controllers/agents/app/serviceclaim_controller.go

```diff
@@ -18,21 +18,39 @@ package controllers
 
 import (
 	"context"
+	"fmt"
 
-	"k8s.io/apimachinery/pkg/runtime"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	primazaiov1alpha1 "github.com/primaza/primaza/api/v1alpha1"
+	sccontrollers "github.com/primaza/primaza/controllers"
+	"github.com/primaza/primaza/pkg/primaza/constants"
 )
 
+const ServiceClaimFinalizer = "serviceclaims.primaza.io/finalizer"
+
+// This is the name of the secret that contains the information the service
+// agents needs to write back registered services up to primaza.  It contains
+// two keys: `kubeconfig`, a serialized kubeconfig for the upstream kubeconfig
+// cluster, and `namespace`, the namespace to write registered services to
+const PRIMAZA_CONTROLLER_REFERENCE string = "primaza-kubeconfig"
+
 // ServiceClaimReconciler reconciles a ServiceClaim object
 type ServiceClaimReconciler struct {
-	client.Client
-	Scheme *runtime.Scheme
+	sccontrollers.ServiceClaimReconciler
 }
 
+//+kubebuilder:rbac:groups=primaza.io,namespace=system,resources=serviceclaims,verbs=get;list;watch;create;update;patch;delete
```

Comment:
These roles are not required here right? As they already exists on primaza controller.